docs: add test to stream_and_receive_listener and mention on json default (#4108)
litestar-org/litestar: docs/usage/websockets.rst

Data can be received in the listener via the ``data`` parameter. The data passed to this
will be converted / parsed according to the given type annotation and supports
:class:`str`, :class:`bytes`, or arbitrary :class:`dict` / or :class:`list` in the
form of JSON.

> The listeners behave exactly like regular handlers, in that they default to JSON, so if the websocket client is expected to send something else, you'll need to explicitly annotate data: str or data: bytes.

Reviewer: ```suggestion
.. important:: 
    
    The listeners will default to JSON unless `data` is annotated with `str` or `bytes`

```
